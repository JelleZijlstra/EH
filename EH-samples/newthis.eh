#!/usr/bin/ehi
class Foo
	private baz = 42
	public bar:
		printvar: $this
		ret $this
	end
	public useprivate:
		ret $baz
	end
end
f := new Foo
printvar: $f->bar:
printvar: $f->useprivate:
echo $baz
