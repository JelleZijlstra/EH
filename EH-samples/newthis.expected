@object <Foo> [
baz <private,non-static,non-constant>: @int 42
useprivate <public,non-static,non-constant>: @function <user>: 
bar <public,non-static,non-constant>: @function <user>: 
]
@object <Foo> [
baz <private,non-static,non-constant>: @int 42
useprivate <public,non-static,non-constant>: @function <user>: 
bar <public,non-static,non-constant>: @function <user>: 
]
@int 42
(null)
